src/dex.hs

```diff
@@ -37,6 +38,26 @@ data EvalMode = ReplMode String
 
 data CmdOpts = CmdOpts EvalMode (Maybe FilePath) EvalConfig
 
+
+
+filteredCompletions keywordList str = map simpleCompletion $
+  filter (str `isPrefixOf`) keywordList
+
+keywordCompletions :: Monad m => CompletionFunc m
+keywordCompletions (line, _) = do
+  -- note: line and thus word and rest have character order reversed
+  let (word, rest) = break (== ' ') line
+  let anywhereKeywords = ["def", "for", "rof", "case", "data", "where", "of", "if", "then", "else", "interface", "instance", "do", "view"]
+  let startoflineKeywords = ["%bench \"", ":p", ":t", ":html", ":export"]
+  let keywords = anywhereKeywords ++ if null rest
+                                        then startoflineKeywords
```

Comment:
haskells indentation rules are really grosss.
idk if there is a nice way to write this without the huge hanging indent